clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -72,13 +76,24 @@
     private static final String ISR_KEY_NAME = "isr";
 
     private final Collection<Node> brokers;
+    private final Node controller;
     private final List<TopicMetadata> topicMetadata;
 
+    /**
+     * Constructor for the latest version
+     */
+    public MetadataResponse(List<Node> brokers, int controllerId, List<TopicMetadata> topicMetadata) {
```

Comment:
Sorry for the confusion. If we don't need to use the older version, we can remove it. However, if we still need to use it, then it's better to keep a constructor for the older version instead of having a version parameter. That's based on Jun's previous reviews, I actually prefer the current approach as we don't need to call deprecated constructors. Anyway, if Gwen is fine with this, maybe we can leave as is.